### PR TITLE
✨ upgrade dependencies

### LINKED ISSUE #8: hatch-pip-compile --upgrade
Currently updating requirements can be tough because of the way the plugin is implemented:

### Current Implementation

1) Create a temporary `requirements.in` file with a project's dependencies
2) Use that `requirements.in` file as the input to `pip-compile` with the output being the respective lock file
3) Since the input is actually a `requirements.in` file and not the actual `pyproject.toml` - some post-processing occurs replacing lines like `# via -r /var/folders/qk/mw5mftjj0qz42nrxrnhctlzw0000gp/T/tmpo85wegym/default.in` with `# via hatch-pip-compile (pyproject.toml)` as well as to add the custom `hatch-pip-compile` header

Because of that implementation there isn't currently a command you could run locally that would update the file in-place. Ideally something like this would just work:
```
hatch dep show requirements --all | \
  hatch run python -m piptools compile - \
    --output-file requirements.txt \
    --generate-hashes \
    --no-header \
    --upgrade-package certifi
```

However running that above command will blow away the `hatch-pip-compile` header.

At this point this issue is just a placeholder and place to brainstorm ideas for the ideal workflow with `pip-compile`

### PR DESCRIPTION
**Pull request by juftin**
closes #8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,6 +53,30 @@ See [lock-filename](#lock-filename) for more information on changing the default
 lockfile paths and [pip-compile-constraint](#pip-compile-constraint) for more
 information on syncing dependency versions across environments.
 
+### Updating Dependencies
+
+Passing arguments like `--upgrade` / `--upgrade-package` to `pip-compile` is
```
Comment (oprypin) [inline README.md:58]: For anyone who doesn't read the whole sentence, the entire takeaway of this is that you can pass an `--upgrade` flag. Which you can't. I suggest not mentioning it at all, or mentioning at the end in parentheses.

File: README.md [hunk 2]
```diff
@@ -53,6 +53,30 @@ See [lock-filename](#lock-filename) for more information on changing the default
...
+set `hatch` will assume that the environment is out of date and automatically
+run `pip-compile` the next time the environment is activated.
+
+The below command runs `pip-compile --upgrade` on the `default` environment.
+
+```shell
+PIP_COMPILE_UPGRADE=1 hatch env run --env default -- python --version
+```
+
+The below command runs `pip-compile --upgrade-package=mkdocs --upgrade-package=mkdocs-material`
```
Comment (oprypin) [inline README.md:70]: Also: me, as a huge fan of `pip-compile`, I've never used `--upgrade-package`. In fact I think it's simply a bad thing to ever update only one package at a time? One package could've broken something and another package may have prepared for it by fixing it in a new release, but you're not getting that fix.

So, this mention could be greatly de-emphasized.

File: README.md [hunk 3]
```diff
@@ -53,6 +53,30 @@ See [lock-filename](#lock-filename) for more information on changing the default
...
+Passing arguments like `--upgrade` / `--upgrade-package` to `pip-compile` is
+controlled by two environment variables: `PIP_COMPILE_UPGRADE` and
+`PIP_COMPILE_UPGRADE_PACKAGE`. When either of these environment variables are
+set `hatch` will assume that the environment is out of date and automatically
+run `pip-compile` the next time the environment is activated.
+
+The below command runs `pip-compile --upgrade` on the `default` environment.
+
+```shell
+PIP_COMPILE_UPGRADE=1 hatch env run --env default -- python --version
```
Comment (oprypin) [inline README.md:67]: Honestly I still think that deleting the lock file is the more ergonomic usage :/

Comment (juftin) [reply README.md]: Fair enough. I mentioned that on the latest update. I think the updated README does a decent job of explaining the upgrade process.

**Review by oprypin — COMMENTED** [review]
Once again great stuff!

I just have a comment about code style, sorry for being nosey

**Pull request merged**